core/server/api/notifications.js

```diff
@@ -121,6 +123,21 @@ notifications = {
         var tasks;
 
         /**
+         * Adds the uuid of notification to "seenNotifications" array.
+         * @param {Object} notification
+         * @return {*|Promise}
+         */
+        function markAsSeen(notification) {
+            var context = {context: {internal: true}};
+            return settings.read({key: 'seenNotifications'}, context).then(function then(response) {
+                var seenNotifications = JSON.parse(response.settings[0].value);
+
+                seenNotifications = _.uniqBy(seenNotifications.concat([notification.uuid]));
+                return settings.edit({settings: [{key: 'seenNotifications', value: seenNotifications}]}, context);
```

Comment:
The args look correct for here. Making the changes above you'd need to change `context` to `{context: context}` though.